trino/serial.go

```diff
@@ -35,6 +35,21 @@ func (e UnsupportedArgError) Error() string {
 // If another string format is used it will error to serialise
 type Numeric string
 
+// Date indicates we want a Date type in Trino from a Golang time.
+type Date time.Time
```

Comment:
OK, so instead of just a type wrapper we make Date and Time more explicit structs with only the fields they need? And keep the time wrappers for timestamps?

Reply:
How they store time internally would be an implementation detail. I guess it only matters if we want to support nano or picoseconds. I was thinking about how to avoid surprises when using them.

Allowing to use `time.Time` directly would be nice too.

Reply:
I have replaced the type wrappers with functions that explicitly request the various fields they contain, or make it explicit what they strip, let me know. I've also removed TimestampTz but added direct support for time.Time (since according to the doc time.Time is ALWAYS carrying a timezone, since nil location means UTC).